lib/service.js

```diff
@@ -143,15 +143,25 @@ class Service extends AdapterService {
   _create (data, params) {
     const discriminator = (params.query || {})[this.discriminatorKey] || this.discriminatorKey;
     const model = this.discriminators[discriminator] || this.Model;
-    return model.create(data).then(result => {
-      if (this.lean) {
-        if (Array.isArray(result)) {
-          return result.map(item => (item.toObject ? item.toObject() : item));
+    if (params.mongoose && params.mongoose.session && typeof data === 'object') {
+      const createOptions = Object.assign({}, params.mongoose);
+      return model.create([data], createOptions)
```

Comment:
Even when you are in a session you could still create array data right? Why not just pass this into the `model.create(data)` that we already have in https://github.com/feathersjs-ecosystem/feathers-mongoose/pull/276/files#diff-6ce8873d3951d1d3483af360fa1f4a93R155

Reply:
Hi @daffl ... I've passed it as a option to create model... 
Note: When a user try creating a single record 'service.create(obj, mongooseOptions)' we need to internally convert it into 'service.create([obj], mongooseOptions)' and them return result[0] because the 'mongoose' dependency does not accept options for a single object creation as it supports a spread of object.